protocols/identify/src/protocol.rs

```diff
@@ -184,7 +219,29 @@ where
     Ok(())
 }
 
-async fn recv<T>(socket: T) -> Result<Info, UpgradeError>
+async fn recv_push<T>(socket: T) -> Result<PushInfo, UpgradeError>
+where
+    T: AsyncRead + AsyncWrite + Unpin,
+{
+    let info = recv(socket).await.and_then(|message| message.try_into())?;
```

Comment:
It would be more idiomatic to `?` after `.await` instead of calling the combinator:

```suggestion
    let info = recv(socket).await?.try_into()?;
```